### DISCUSSION TITLE
how to change ui.aggrid row color for every single row?

### DISCUSSION THREAD

**Discussion by zhaoguotao** [zauberzeug/nicegui#2455] [Q&A]
### Question

I want to change one row's color for ui.aggrid, how can i achieve that?

**Comment by falkoschindler** [comment]
Hi @zhaoguotao,

You can customize the background color of grid rows using AG Grid's [`getRowStyle`](https://www.ag-grid.com/javascript-data-grid/row-styles/) function:

```py
ui.aggrid({
    'defaultColDef': {'flex': 1},
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'},
    ],
    'rowData': [
        {'name': 'Alice', 'age': 18},
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol', 'age': 42},
    ],
    ':getRowStyle': '(params) => params.data.age >= 21 ? { background: "lightgreen" } : null',
})
```

![Screenshot 2024-01-29 at 15 10 09](https://github.com/zauberzeug/nicegui/assets/5767091/5aab1c70-de77-4136-84be-a71380ebc389)
